bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostCB.scala

```diff
@@ -89,64 +89,70 @@ final class GhostCB(obsComp: ISPObsComponent) extends AbstractObsComponentCB(obs
       getObsComponent.getContextObservation.getObsComponents.asScala.
         find(_.getType.broadType === TargetObsComp.SP_TYPE.broadType).
         map(_.getDataObject.asInstanceOf[TargetObsComp]).
-        foreach{_.getAsterism match {
-
-          /** STANDARD RESOLUTION
-           * 1. If the base is overridden, add it to the parameters.
-           * 2. If SRIFU1 is pointing to a sky position, write sky position coordinates.
-           *    Otherwise, write target coordinates and indicate if guiding is turned on for SRIFU1.
-           * 3. If SRIFU2 is set, it can be a sky position or target: repeat the previous step. for SRIFU2 data.
-           */
-          case gsr: GhostAsterism.StandardResolution =>
-            gsr.overriddenBase.foreach(b => coordParam(b, None,
-              Ghost.BASE_RA_DEGREES, Ghost.BASE_DEC_DEGREES,
-              Ghost.BASE_RA_HMS, Ghost.BASE_DEC_DMS))
-
-            gsr.srifu1.fold(c => coordParam(c, Some(Ghost.SRIFU1_NAME),
-              Ghost.SRIFU1_RA_DEG, Ghost.SRIFU1_DEC_DEG,
-              Ghost.SRIFU1_RA_HMS, Ghost.SRIFU1_DEC_DMS),
-              t => {
-                coordParam(t.spTarget, Some(Ghost.SRIFU1_NAME),
-                  Ghost.SRIFU1_RA_DEG, Ghost.SRIFU1_DEC_DEG,
-                  Ghost.SRIFU1_RA_HMS, Ghost.SRIFU1_DEC_DMS)
-                guiding(Ghost.SRIFU1_GUIDING, t)
-              })
-
-            gsr.srifu2.foreach(_.fold(c => coordParam(c, Some(Ghost.SRIFU2_NAME),
-              Ghost.SRIFU2_RA_DEG, Ghost.SRIFU2_DEC_DEG,
-              Ghost.SRIFU2_RA_HMS, Ghost.SRIFU2_DEC_DMS),
-              t => {
-                coordParam(t.spTarget, Some(Ghost.SRIFU2_NAME),
-                  Ghost.SRIFU2_RA_DEG, Ghost.SRIFU2_DEC_DEG,
-                  Ghost.SRIFU2_RA_HMS, Ghost.SRIFU2_DEC_DMS)
-
-                guiding(Ghost.SRIFU2_GUIDING, t)
-              }))
-
-            /** HIGH RESOLUTION
+        foreach{t =>
+          t.getTargetEnvironment.getUserTargets.asScala.zipWithIndex.foreach { case (ut, i) =>
+            val (a, b, c, d, e)= Ghost.userTargetParams(i + 1)
+            coordParam(ut.target, Some(a), b, c, d, e)
+            config.putParameter(systemName, DefaultParameter.getInstance(s"userTarget${i+1}Type", ut.`type`))
+          }
```

Comment:
This is really the new part correct?  The rest is just moved down from the previous version?

Reply:
The diff got a bit strange because the old part was moved down. It seems fine to me as the OT shows the old configuration items as well as the new ones